src/Servers/Kestrel/Core/src/Internal/Http2/Http2MessageBody.cs

```diff
@@ -111,10 +111,23 @@ public override void AdvanceTo(SequencePosition consumed, SequencePosition exami
 
         public override bool TryRead(out ReadResult readResult)
         {
-            var result = _context.RequestBodyPipe.Reader.TryRead(out readResult);
-            _readResult = readResult;
+            TryStart();
```

Comment:
Was this just a bug you noticed and are fixing here too?

Reply:
Yes all the timing and bandwidth write tests fail if you don't go via ReadAsync as there is nothing to trigger them in `TryRead`

Reply:
Well, not all, some :)